### PR TITLE
Assorted code cleanups, part 2

### PR DESCRIPTION
**Pull request by kholia**
This is related to https://github.com/magnumripper/JohnTheRipper/issues/3001.

@magnumripper Do the OpenMP related changes look OK to you?

Instead of having a single big commit, I will add multiple commits to this PR. This should making reviewing things a bit easier.

### CODE REVIEW

**Comment by kholia** [comment]
I don't like the OpenMP usage in `cmp_all`. It works but the construction looks suspect. I am applying the following patch,

``` diff
diff --git a/src/eigrp_fmt_plug.c b/src/eigrp_fmt_plug.c
index b2e976030..faa7f7b49 100644
--- a/src/eigrp_fmt_plug.c
+++ b/src/eigrp_fmt_plug.c
@@ -310,9 +312,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 static int cmp_all(void *binary, int count)
 {
        int index = 0;
-#ifdef _OPENMP
+
        for (; index < count; index++)
-#endif
                if (((uint32_t*)binary)[0] == crypt_out[index][0])
                        return 1;
        return 0;
```

File: src/eigrp_fmt_plug.c [hunk 1]
```diff
@@ -274,8 +276,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	int index = 0;
 #ifdef _OPENMP
 #pragma omp parallel for
-	for (index = 0; index < count; index++)
 #endif
+	for (index = 0; index < count; index++)
 	{
```
Comment (solardiz) [inline src/eigrp_fmt_plug.c]: The bracket should go on the line with `for`. The only reason why it was on its own line was because of the `#ifdef`.

File: src/eigrp_fmt_plug.c [hunk 2]
```diff
@@ -310,9 +312,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 static int cmp_all(void *binary, int count)
 {
 	int index = 0;
-#ifdef _OPENMP
+
 	for (; index < count; index++)
```
Comment (solardiz) [inline src/eigrp_fmt_plug.c]: `index = 0` should be in the `for`

File: src/skein_fmt_plug.c [hunk 3]
```diff
@@ -225,21 +228,23 @@ static int crypt_512(int *pcount, struct db_salt *salt)
...
 	return count;
 }
 
 static int cmp_all(void *binary, int count)
 {
 	int index = 0;
+
 #ifdef _OPENMP
 	for (; index < count; index++)
 #endif
```
Comment (solardiz) [inline src/skein_fmt_plug.c]: Forgot to delete the `#ifdef` logic here

File: src/eigrp_fmt_plug.c (hunk 1 above)
Comment (kholia) [reply src/eigrp_fmt_plug.c]: They do look odd on a new line. Will fix.

This whole idea of mine to have separate commits is not working well from a CI perspective. The CI is trying to build each commit individually (quite logical) which is taking a huge amount of time.

**Comment by kholia** [comment]
I will add more changes to this PR tomorrow.

**Comment by magnumripper** [comment]
8352.6
```
citrix_ns_fmt_plug.c:284:26: error: use of undeclared identifier 'loops'
        for (index = 0; index < loops; ++index) {
                                ^
```

**Comment by magnumripper** [comment]
Original code
```c
#ifdef _OPENMP
	int loops = (count + MAX_KEYS_PER_CRYPT - 1) / MAX_KEYS_PER_CRYPT;

#pragma omp parallel for
	for (index = 0; index < loops; ++index)
#endif
```
You need to drop the `#ifdef`

**Comment by kholia** [comment]
Done now, thanks!

**Comment by kholia** [comment]
I am done with changes for now. This PR should be ready for another round of review.

File: src/aix_smd5_fmt_plug.c [hunk 4]
```diff
@@ -319,20 +319,19 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
 }
 
 static int cmp_all(void *binary, int count)
 {
 	int index = 0;
-#ifdef _OPENMP
-	for (; index < count; index++)
-#endif
+
+	for (index = 0; index < count; index++)
```
Comment (solardiz) [inline src/aix_smd5_fmt_plug.c]: Duplicate `index = 0`

File: src/citrix_ns_fmt_plug.c [hunk 5]
```diff
@@ -221,24 +221,24 @@ static void set_salt(void *salt)
 static int cmp_all(void *binary, int count)
 {
 #ifdef SIMD_COEF_32
-	unsigned int x, y=0;
+	unsigned int x, y = 0;
 
-	for (; y < kpc/SIMD_COEF_32; y++)
-		for (x = 0; x < SIMD_COEF_32; x++)
-		{
+	for (; y < kpc/SIMD_COEF_32; y++) {
```
Comment (solardiz) [inline src/citrix_ns_fmt_plug.c]: Should move the `y = 0` to `for`.

File: src/django_scrypt_fmt_plug.c [hunk 6]
```diff
@@ -188,27 +192,25 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
-	{
+	for (index = 0; index < count; index++) {
 		if (crypto_scrypt((unsigned char*)saved_key[index], strlen((char*)saved_key[index]),
 				cur_salt->salt, strlen((char*)cur_salt->salt),
 				(1ULL) << cur_salt->N, cur_salt->r,
 				cur_salt->p, (unsigned char*)crypt_out[index],
-				BINARY_SIZE) == -1)
-		{
+				BINARY_SIZE) == -1) {
 			memset(crypt_out[index], 0, sizeof(crypt_out[index]));
```
Comment (solardiz) [inline src/django_scrypt_fmt_plug.c]: I think we shouldn't silently ignore errors from scrypt. They are not same as a wrong password. It'd be more appropriate to print a message and maybe even `error()` out afterwards. But this is not a cleanup matter, so should be in a separate PR, and perhaps not limited to this one instance of the problem.

File: src/gost_fmt_plug.c [hunk 7]
```diff
@@ -203,13 +202,15 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
+
 	return count;
 }
 
 static int cmp_all(void *binary, int count)
 {
 	int index = 0;
-	for (; index < count; index++)
+
+	for (index = 0; index < count; index++)
```
Comment (solardiz) [inline src/gost_fmt_plug.c]: Duplicate `index = 0`

File: src/hsrp_fmt_plug.c [hunk 8]
```diff
@@ -10,7 +10,7 @@
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted.
  *
- * optimized Feb 2016, JimF.
+ * Pptimized in Feb 2016, JimF.
```
Comment (solardiz) [inline src/hsrp_fmt_plug.c]: Typo

File: src/hsrp_fmt_plug.c [hunk 9]
```diff
@@ -228,15 +230,15 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
 }
 
 static int cmp_all(void *binary, int count)
 {
 	int index = 0;
-#ifdef _OPENMP
-	for (; index < count; index++)
-#endif
+
+	for (index = 0; index < count; index++)
```
Comment (solardiz) [inline src/hsrp_fmt_plug.c]: Duplicate `index = 0`

File: src/keychain_fmt_plug.c [hunk 10]
```diff
@@ -206,9 +206,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	int index = 0;
```
Comment (solardiz) [inline src/keychain_fmt_plug.c]: Duplicate `index = 0`. Probably the same problem exists in many other of the patched `crypt_all()`, but just happened to be out of the patch context seen by default, thus I didn't notice it there.

File: src/krb5pa-sha1_fmt_plug.c [hunk 11]
```diff
@@ -329,9 +329,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	int index = 0;
```
Comment (solardiz) [inline src/krb5pa-sha1_fmt_plug.c]: Duplicate `index = 0`.

File: src/krb5pa-md5_fmt_plug.c [hunk 12]
```diff
@@ -358,9 +357,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 
 #ifdef _OPENMP
 #pragma omp parallel for
-	for (i = 0; i < count; i++)
 #endif
-	{
+	for (i = 0; i < count; i++) {
```
Comment (solardiz) [inline src/krb5pa-md5_fmt_plug.c]: Duplicate `i = 0` closer to the beginning of the function

File: src/lastpass_sniffed_fmt_plug.c [hunk 13]
```diff
@@ -187,9 +187,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	int index = 0;
```
Comment (solardiz) [inline src/lastpass_sniffed_fmt_plug.c]: Duplicate `index = 0`

File: src/salted_sha1_fmt_plug.c [hunk 14]
```diff
@@ -215,18 +217,16 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
-	int inc = 1;
+	inc = NBKEYS;
 #endif
 
+#ifdef _OPENMP
 #pragma omp parallel for
-	for (index=0; index < count; index += inc)
 #endif
-	{
+	for (index=0; index < count; index += inc) {
```
Comment (solardiz) [inline src/salted_sha1_fmt_plug.c]: Looks like `inc` is a constant, so keeping it in a variable which you first set to 1 and then override to NBKEYS is weird. Even the original code (prior to this patch) wasn't nearly as weird. Also, duplicate `index = 0` above, and missing spaces around the `=`.

File: src/sapG_fmt_plug.c [hunk 15]
```diff
@@ -458,8 +458,7 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 #if defined(_OPENMP)
 #pragma omp parallel for
 #endif
-	for (t = 0; t < (count-1)/(NBKEYS)+1; t++)
-	{
+	for (t = 0; t < (count-1)/(NBKEYS)+1; t++) {
```
Comment (solardiz) [inline src/sapG_fmt_plug.c]: We generally use signed int for the index variable in OpenMP `for` loops, because support for unsigned requires OpenMP 3.0 spec support from the compiler, IIRC. Maybe we depend on that spec (or higher?) anyway, but at least the core tree tries to target OpenMP 2.5+.

File: src/skein_fmt_plug.c [hunk 16]
```diff
@@ -206,15 +208,16 @@ static int crypt_256(int *pcount, struct db_salt *salt)
 
 #ifdef _OPENMP
 #pragma omp parallel for
-	for (index = 0; index < count; index++)
 #endif
+	for (index = 0; index < count; index++)
 	{
```
Comment (solardiz) [inline src/skein_fmt_plug.c]: Brace placement. Also duplicate `index = 0` as usual.

File: src/sl3_fmt_plug.c [hunk 17]
```diff
@@ -255,18 +256,16 @@ static int crypt_all(int *pcount, struct db_salt *salt)
...
-	int inc = 1;
+	inc = NBKEYS;
 #endif
 
+#ifdef _OPENMP
 #pragma omp parallel for
-	for (index=0; index < count; index += inc)
 #endif
-	{
+	for (index=0; index < count; index += inc) {
```
Comment (solardiz) [inline src/sl3_fmt_plug.c]: Same comments as I made on another format with similar patterns

**Comment by kholia** [comment]
@solardiz Thanks for pointing out all these problems. I am fixing them now.

I am also fixing things found by `grep -Rin "int index = 0" -A 5 | grep "for (index ="`.

**Comment by kholia** [comment]
Another grep query -> `grep -Rin 'for (; index < count; index++)'`.

File: src/sapG_fmt_plug.c (hunk 15 above)
Comment (kholia) [reply src/sapG_fmt_plug.c]: This should be fixed now.

```
@@ -454,12 +448,11 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 #if SIMD_COEF_32
 #define ti (t*NBKEYS+index)
 
-       unsigned t;
+       int t;
 #if defined(_OPENMP)
```

File: src/krb5pa-md5_fmt_plug.c (hunk 12 above)
Comment (kholia) [reply src/krb5pa-md5_fmt_plug.c]: Fixed now.

**Comment by kholia** [comment]
I am not sure about my removal of `#if defined(_OPENMP) || MAX_KEYS_PER_CRYPT > 1` guard.

File: src/django_scrypt_fmt_plug.c (hunk 6 above)
Comment (kholia) [reply src/django_scrypt_fmt_plug.c]: Ack.

**Comment by magnumripper** [comment]
> I am not sure about my removal of `#if defined(_OPENMP) || MAX_KEYS_PER_CRYPT > 1` guard. This can affect the generated code a bit.

I can't see any such thing removed right now? However, @solardiz mentioned somewhere that such conditionals should probably be removed and (if needed) MAX_KEYS_PER_CRYPT should be bumped/tuned for non-OMP. After that, OMP_SCALE can be (lowered and) tuned for OMP.

**Comment by kholia** [comment]
> I can't see any such thing removed right now? 

I reverted my changes but forgot to update (inform) this thread. My bad.

I will check how removing this guard affects the performance in non-OpenMP builds.

**Comment by kholia** [comment]
I am planning to address the `#if defined(_OPENMP) || MAX_KEYS_PER_CRYPT > 1` construction in a new PR.

**Comment by kholia** [comment]
This should be OK to merge. Merging!
